packages/kbn-shared-ux-components/src/page_template/solution_nav/solution_nav.tsx

```diff
@@ -67,68 +102,94 @@ export const KibanaPageTemplateSolutionNav: FunctionComponent<
-      name={name}
-    />
-  ) : null;
+  const solutionAvatar = useMemo(() => {
+    if (!icon) {
+      return;
+    }
+    return (
+      <KibanaSolutionAvatar
+        className="kbnPageTemplateSolutionNav__avatar"
+        iconType={icon}
+        name={name}
+      />
+    );
+  }, [name, icon]);
 
   /**
    * Create the titles
    */
-  const titleText = (
-    <>
-      {solutionAvatar}
-      <strong>{name}</strong>
-    </>
-  );
-  const mobileTitleText = (
-    <FormattedMessage
-      id="sharedUXComponents.solutionNav.mobileTitleText"
-      defaultMessage="{solutionName} Menu"
-      values={{ solutionName: name || 'Navigation' }}
-    />
+  const headingID = headingProps?.id || generateId('heading');
+  const HeadingElement = headingProps?.element || 'h2';
+  const titleText = useMemo(
+    () => (
+      <EuiTitle size="xs" id={headingID}>
+        <HeadingElement>
+          {solutionAvatar}
+          <strong>
+            <FormattedMessage
+              id="sharedUXComponents.solutionNav.mobileTitleText"
+              defaultMessage="{solutionName} {menuText}"
+              values={{
+                solutionName: name || 'Navigation',
+                menuText: isSmallerBreakpoint
+                  ? i18n.translate('sharedUXComponents.solutionNav.menuText', {
+                      defaultMessage: 'menu',
+                    })
+                  : '',
+              }}
+            />
+          </strong>
+        </HeadingElement>
+      </EuiTitle>
+    ),
+    [name, solutionAvatar, headingID, HeadingElement, isSmallerBreakpoint]
   );
 
   /**
-   * Create the side nav component
+   * Create the side nav content
    */
-
-  const sideNav = () => {
+  const sideNavContent = useMemo(() => {
+    if (isCustomSideNav) {
+      return children;
+    }
     if (!items) {
       return null;
     }
-    const sideNavClasses = classNames('kbnPageTemplateSolutionNav', {
-      'kbnPageTemplateSolutionNav--hidden': isHidden,
-    });
     return (
       <EuiSideNav
+        aria-labelledby={headingID}
         aria-hidden={isHidden}
-        className={sideNavClasses}
-        heading={titleText}
-        mobileTitle={
-          <>
-            {solutionAvatar}
-            {mobileTitleText}
-          </>
-        }
-        toggleOpenOnMobile={toggleOpenOnMobile}
-        isOpenOnMobile={isSideNavOpenOnMobile}
         items={setTabIndex(items, isHidden)}
+        mobileBreakpoints={[]} // prevent EuiSideNav to apply mobile version, already implemented here
         {...rest}
       />
     );
-  };
+  }, [children, headingID, isCustomSideNav, isHidden, items, rest]);
 
   return (
     <>
-      {isSmallerBreakpoint && sideNav()}
+      {isSmallerBreakpoint && (
+        <EuiCollapsibleNavGroup
+          className={sideNavClasses}
+          paddingSize="m"
+          background="none"
+          title={titleText}
+          titleElement="span"
+          isCollapsible={true}
+          initialIsOpen={false}
```

Comment:
I got a second opinion on the `apps` icon and we've decided to remove it mainly because it conflicts with the top menu icon of the same. Also, I updated the mobile menu text since it was missing the word `menu`